### PR TITLE
Add support for the SMSPlus-GX core

### PR DESCRIPTION
**Pull request by michael-j-green**
This PR adds the SMSPlus-GX core, which seems to more reliably play Sega Master System and Game Gear ROMs, than the current Genesis core.

This new core will be the default used when the segaMS alias is used.

Also added a query string parameter to enable threads: ``http://url/index.html?threads=1``

Threads will normally be disabled

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -107,16 +107,26 @@ <h1>EmulatorJS Demo</h1>
 
         <script>
             var enableDebug = false;
+            var enableThreads = false;
```
Comment (ethanaobrien) [inline index.html]: Could we use `let` here

File: index.html [hunk 2]
```diff
@@ -107,16 +107,26 @@ <h1>EmulatorJS Demo</h1>
...
+            if (urlParams.get('threads') == 1)
+                enableThreads = true;
             
             if (enableDebug == true) {
                 console.log("Debug is enabled");
             } else {
                 console.log("Debug is disabled");
             }
+
+            if (enableThreads == true) {
```
Comment (ethanaobrien) [inline index.html]: No need for the ` == true`

File: index.html [hunk 3]
```diff
@@ -107,16 +107,26 @@ <h1>EmulatorJS Demo</h1>
...
         <script>
             var enableDebug = false;
+            var enableThreads = false;
             const queryString = window.location.search;
             const urlParams = new URLSearchParams(queryString);
             if (urlParams.get('debug') == 1)
                 enableDebug = true;
+
+            if (urlParams.get('threads') == 1)
+                enableThreads = true;
```
Comment (ethanaobrien) [inline index.html]: Can we also assert the existence of `SharedArrayBuffer`

**Review by ethanaobrien — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (michael-j-green) [reply index.html]: Both enableDebug and enableThreads have now been changed to let

File: index.html (hunk 2 above)
Comment (michael-j-green) [reply index.html]: Simplified the code by not doing the same test twice.

**Review by ethanaobrien — APPROVED** [review]

**Pull request merged**